src/operations/semester.ts

```diff
@@ -0,0 +1,72 @@
+import { SemesterCreate, SemesterUpdate, Semester } from '../models'; // Replace with the correct import path for your interfaces
+import getDb from '../initDB'; // Replace with the correct import path for your getDb function
+
+const prisma = getDb();
+
+export async function createSemester(data: SemesterCreate): Promise<Semester> {
+  return prisma.semester.create({
+    data: {
+      name: data.name,
+      ownerId: data.ownerId,
+    },
+  });
+}
+
+export async function getSemesterById(
+  id: number,
+  ownerId: string
+): Promise<Semester> {
+  try {
+    const semester = await prisma.semester.findFirst({
+      where: { id, ownerId },
+    });
+    if (!semester) throw new Error('');
+    return semester;
+  } catch (e) {
+    throw new Error('Semester not found');
+  }
+}
+
+export async function getSemestersByOwnerId(
+  ownerId: string
+): Promise<Semester[]> {
+  try {
+    return prisma.semester.findMany({
+      where: { ownerId },
+    });
+  } catch (e) {
+    throw new Error('Semesters not found');
+  }
+}
+
+export async function updateSemester(
+  id: number,
+  ownerId: string,
+  data: SemesterUpdate
+): Promise<Semester> {
+  try {
+    const semester = await getSemesterById(id, ownerId);
+    return prisma.semester.update({
```

Comment:
I'm curious what the reason is for splitting this up into two queries because to me it seems like it can be done in just one. Is it cleaner to do it this way?